gnomad_qc/v3/annotations/generate_freq_data.py

```diff
@@ -59,16 +61,36 @@ def make_faf_index_dict(faf_meta: List[Dict[str, str]]) -> Dict[str, int]:
 
 
 def main(args):
-    subset = args.subset
+    subsets = args.subsets
     hl.init(
-        log=f"/generate_frequency_data{'.' + subset if subset else ''}.log",
+        log=f"/generate_frequency_data{'.' + '_'.join(subsets) if subsets else ''}.log",
         default_reference="GRCh38",
     )
 
+    invalid_subsets = []
+    n_subsets_in_use_subpops = 0
+    for s in subsets:
+        if s not in SUBSETS:
+            invalid_subsets.append(s)
+        if s in COHORTS_WITH_POP_STORED_AS_SUBPOP:
+            n_subsets_in_use_subpops += 1
+
+    if len(invalid_subsets) > 0:
+        raise ValueError(
+            f"{', '.join(invalid_subsets)} subset(s) are not one of the following official subsets: {SUBSETS}"
+        )
+    if (n_subsets_in_use_subpops > 0) & (n_subsets_in_use_subpops != len(subsets)):
+        raise ValueError(
+            f"All or none of the supplied subset(s) should be in the list of cohorts that need to use subpops instead "
+            f"of pops in frequency calculations: {COHORTS_WITH_POP_STORED_AS_SUBPOP}"
+        )
+
     try:
         logger.info("Reading full sparse MT and metadata table...")
         mt = get_gnomad_v3_mt(
-            key_by_locus_and_alleles=True, release_only=True, samples_meta=True
+            key_by_locus_and_alleles=True,
```

Comment:
Actually, I think this is a very good question. I think overall for the `get_gnomad_v3_mt` function this is probably fine because it prevents needing to do the # AC any everytime and add in the filter in cases where it's not needed. In downstream analyses in this script there is the addition of the frequency annotation and 
```
            # Remove all loci with raw AC=0
            mt = mt.filter_rows(mt.freq[1].AC > 0)
```
happens later. 

However, the question is do we need to keep these variants for the subsets. as of now we are setting the subset variants to null structs in the frequency anyway, so after the subset frequency we might also save space by removing these.